### DISCUSSION TITLE
생각해 볼 내용

### DISCUSSION THREAD

**Discussion by flowjiyun** [Webwaiter/Webwaiter#51] [Ideas]
- 기본 내장 함수 예외 던져졌을 때 서버 안죽도록 만들어야함..

**Comment by kimchijinju** [comment]
- 메서드 판단하는 코드 필요
- Host 필드가 없을 경우 bad request 처리

**Reply by kimchijinju** [reply]
완료

**Comment by flowjiyun** [comment]
cgi process의 path를 location block 안으로 이동 : 여러 cgi process 실행을 위해

**Reply by kimchijinju** [reply]
완료

**Comment by flowjiyun** [comment]
구현 이슈
- [x] chuncked message일때, 파싱할 수 있는 만큼 전부 파싱하지 않고 1개의 청크만 파싱하고 i/o loop을 거쳐서 다시 들어오는 상황
- [x] if modified since 할지 말지
- [x] request body size 와 client max body 사이즈 비교해서 올바른 요청인지 확인하는 로직 추가
- [x] telnet에서 crlf 보내면 스킵 로직
- [x] method는 무조건 대문자로만 와야 하는 건지 확인하고 에러코드 변경
- [x] 파일 업로드 파이선 스크립트 만들고 요청 메세지도 확인
- [x] sigpipe 생겼을 때 서버 죽는 문제 해결
- [x] pipe_writeing_fd 닫는 시점
- [x] .cgi 확장자일 경우 location block 선택하는 로직

테스트
- [x] 파일업로드 테스트
- [x] 파일 삭제 테스트
- [x] 무한 루프 스크립트 테스트
- [x] 메모리 릭 테스트
- [x] seige
- [x] client bodysize 테스트 
- [x] 여러 포트, 동일포트 여러 호스트
- [x] 리다이렉션 테스트
- [x] 디렉토리 리스팅 테스트
